core/src/main/java/org/apache/calcite/sql/validate/SqlValidatorUtil.java

```diff
@@ -1326,6 +1327,48 @@ public static void adjustTypeForArrayConstructor(
     }
   }
 
+  /**
+   * When the array_append and array_prepend element does not equal the array component type
+   * make explicit casting.
+   *
+   * @param componentType derived array component type
+   * @param opBinding description of call
+   */
+  public static void adjustTypeForArrayFunctionConstructor(
```

Comment:
I think here you need to pass 3 args. 
arrayComponentType/elementType/targetType.

Reply:
removeit